core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java

```diff
@@ -0,0 +1,345 @@
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.*;
+import io.questdb.std.Long256;
+import io.questdb.std.Long256Impl;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+import io.questdb.std.str.CharSink;
+
+public class CoalesceFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "coalesce(V)";
+    }
+
+    @Override
+    public Function newInstance(ObjList<Function> args, int position, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        // Similar to CASE function.
+        // compute return type in this loop
+        final int argsSize = args.size();
+        int returnType = -1;
+        for (int i = 0; i < argsSize; i++) {
+            returnType = CaseCommon.getCommonType(returnType, args.getQuick(i).getType(), args.getQuick(i).getPosition());
+        }
+
+        switch (returnType) {
+            case ColumnType.DOUBLE:
+                return new DoubleCoalesceFunction(position, args);
+            case ColumnType.DATE:
+                return new DateCoalesceFunction(position, args);
+            case ColumnType.TIMESTAMP:
+                return new TimestampCoalesceFunction(position, args);
+            case ColumnType.LONG:
+                return new LongCoalesceFunction(position, args);
+            case ColumnType.LONG256:
+                return new Long256CoalesceFunction(position, args);
+            case ColumnType.INT:
+                return new IntCoalesceFunction(position, args);
+            case ColumnType.FLOAT:
+                return new FloatCoalesceFunction(position, args);
+            case ColumnType.STRING:
+            case ColumnType.SYMBOL:
+                return new SymStrCoalesceFunction(position, args);
+            case ColumnType.BOOLEAN:
+            case ColumnType.SHORT:
+            case ColumnType.BYTE:
+            case ColumnType.CHAR:
+                // Null on these data types not supported
+                return args.getQuick(0);
+            default:
+                throw SqlException.$(position, "coalesce cannot be used with ")
+                        .put(ColumnType.nameOf(returnType))
+                        .put(" data type");
+        }
+    }
+
+    private static class DoubleCoalesceFunction extends DoubleFunction implements MultiArgFunction {
+        private final ObjList<Function> args;
+
+        public DoubleCoalesceFunction(int position, ObjList<Function> args) {
+            super(position);
+            this.args = args;
+        }
+
+        @Override
+        public ObjList<Function> getArgs() {
+            return args;
+        }
+
+        @Override
+        public double getDouble(Record rec) {
+            for (int i = 0, size = args.size(); i < size; i++) {
```

Comment:
`args.size()` should not be evaluated for each record

Reply:
done